test/dynamo/test_higher_order_ops.py

```diff
@@ -417,6 +430,135 @@ def f(x):
         self._test_wrap_simple(f, (x,), 3, expected_opcount=2)
 
 
+class ActivationCheckpointingTests(torch._dynamo.test_case.TestCase):
+    def _validate(self, fn, backend, *args, skip_check=False, fullgraph=True):
+        expected = fn(*args)
+        expected.sum().backward()
+
+        result = torch.compile(fn, fullgraph=fullgraph, backend=backend)(*args)
+        result.sum().backward()
+
+        if not skip_check:
+            self.assertEqual(result, expected)
```

Comment:
Do we want to check that the gradients are the same? Maybe by using torch.autograd.grad instead of .backward()?

Reply:
Added